src/tests/ftest/pool/dynamic_server_pool.py

```diff
@@ -0,0 +1,152 @@
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from apricot import TestWithServers
+from test_utils_pool import TestPool
+from check_for_pool import check_for_pool
+
+
+class DynamicServerPool(TestWithServers):
+    """Test Class Description:
+    Test to create pools on single, multiple, and dynamically added server.
+
+    This test covers the following requirements.
+    SRS-10-0054: New server can be used for new pool.
+    SRS-10-0105: A pool spanning only a subset of the servers can be created
+    starting from 1 server up to all the servers. Query should be used to verify
+    that the pool was allocated on the requested servers.
+
+    First, we'll create one pool on the first server and another pool over two
+    servers. This covers SRS-10-0105.
+
+    Second, we start a new server and let it join the existing two servers.
+    Then we'll create one pool on this new server and another one over all three
+    servers. This covers SRS-10-0054.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DynamicServerPool object."""
+        super(DynamicServerPool, self).__init__(*args, **kwargs)
+        self.expected_uuids = []
+
+    def create_new_pool(self, ranks=None):
+        """Create a new pool using ranks and append it to self.pool.
+
+        Args:
+            ranks (str, optional): Ranks value passed in to --ranks. Defaults to
+            None.
+        """
+        new_pool = TestPool(self.context, dmg_command=self.get_dmg_command())
+        new_pool.get_params(self)
+        if ranks:
+            new_pool.target_list.update(ranks)
+        new_pool.create()
+        self.expected_uuids.append(new_pool.uuid.lower())
+        self.pool.append(new_pool)
+
+    def verify_uuids(self):
+        """Call dmg pool list to get the list of UUIDs and verify against the
+        expected UUIDs passed in.
+        """
+        actual_uuids = []
+        output_list = self.get_dmg_command().get_output("pool_list")
+        for uuid_sr in output_list:
+            actual_uuids.append(uuid_sr[0])
+        actual_uuids.sort()
+        self.log.debug(str(actual_uuids))
```

Comment:
A message would be useful here to know what is being printed.

Reply:
I forgot to remove this debug line. Thanks for catching.